vignettes/atlascreation.Rmd

```diff
@@ -0,0 +1,143 @@
+
+GSE124952Normalized <- NormalizeData(mat_PFC)
+GSE124952Normalized
+
+new_ref_matrix <- average_clusters(mat = GSE124952Normalized, metadata = meta_PFC$CellType, if_log = FALSE)
+saveRDS(new_ref_matrix, "GSE124952.rds")
+```
+
+## Concatenating Reference Matrices into Cell Atlas
+
+Once the individual reference matrices are generated, they can be concatenated into a single cell atlas encompassing all cells in an organism. In this case, all of the reference matrices are stored as RDS files in separate folder. In order to combine them, the union of all genes from a mice organism and genes present in the reference matrices will be applied. This will ensure that the reference matrices are all of the same row length which will make them easy to combine with a function such as `cbind()`. The union will be found with the `append_genes()` function and the atlas will be built with the `build_atlas()` function. 
+
+The `append_genes()` function inputs a character vector with the list of all mouse genes and an average gene expression reference matrix. The row names of the average gene expression matrix will be found using `rownames()` R function. In order to figure out which gene names are missing from the reference matrix, the `setdiff()` function will be utilized. All of the missing genes will be added to a zero reference matrix (all of the missing genes will be assigned average gene expression values of zero). The zero expression matrix and the reference matrix will then be merged with the `rbind()` function. A final re-ordering of gene names will be done and then the full matrix will be returned. The output should be a reference matrix with the same number of rows as the mouse gene character vector. 
+
+The `build_atlas()` function inputs a character vector of paths to study matrices stored as .rds files (matrix_fns), a text file with a single column containing genes and the ordering desired in the output matrix (genes_fn) and a constant output filename. In the function, it is set to a default `NULL`. The contents of gene_fn is read into a gene name character vector for later use. The reference matrices are then read with the help of matrix_fns and `readRDS()` (reference matrices can be stored as .rds files). Over the list of collected reference matrices, the `append_genes()` function will be run which will elongate all reference matrices into a reference matrix that has the same number of rows as the mouse gene vector. The relevant study names for each cell type column name will then be pasted and the reference matrices will be concatenated into a single gene expression atlas matrix. The matrix will then be stored in a separate directory called "atlas".   
+
+The following code will demonstrate the concatenation of the GSE113049 and GSE124952 reference matrices into a cell atlas. 
+```{r Building atlas with reference matrices}
+library(dplyr)
+library(Seurat)
+library(clustifyr)
+library(tidyverse)
+library(readr)
+library(digest)
+library(here)
```

Comment:
by not using `here`, please adapt the code to 
1, use case where we just want to concat 2 matrices already in the enviroment
2, in a chunk that doesn't run, (eval = F), give a hypothetical scenario of how going directly from rds files would work